Cabal/src/Distribution/Simple/Configure.hs

```diff
@@ -935,6 +956,34 @@ configurePackage cfg lbc0 pkg_descr00 flags enabled comp platform programDb0 pac
 
   return (lbc, pbd)
 
+nameOf :: LegacyExeDependency -> String
+nameOf (LegacyExeDependency n _) = n
+
+versionOf :: LegacyExeDependency -> VersionRange
+versionOf (LegacyExeDependency _ v) = v
+
+versionOfDoc :: LegacyExeDependency -> Doc
+versionOfDoc (LegacyExeDependency _ v) =
+  if v == anyVersion
+    then text (prettyShow v) <+> parens (text "any version")
+    else text $ prettyShow v
+
+-- | Any duplicates in the list has their version range merged by intersection.
+-- The second list has the build tool with its merged version range and its list
+-- of duplicates.
+deduplicateBuildTools :: [LegacyExeDependency] -> ([LegacyExeDependency], [(LegacyExeDependency, [LegacyExeDependency])])
+deduplicateBuildTools xs =
+  catMaybes
+    <$> unzip
+      [ (merged, if length gs == 1 then Nothing else Just (merged, gs))
+      | gs@(g : _) <- groupBy ((==) `on` nameOf) (sortBy (comparing nameOf) xs)
+      , let merged = LegacyExeDependency (nameOf g) (mergeVersions (ordNub . filter (/= anyVersion) $ versionOf <$> gs))
+      ]
+  where
+    mergeVersions :: [VersionRange] -> VersionRange
+    mergeVersions [] = anyVersion
+    mergeVersions (v : vs) = foldr intersectVersionRanges v vs
```

Comment:
`foldx intersectVersionRanges` seems a common pattern, maybe this version should be in `Cabal-syntax`?

See: `cabal-install/src/Distribution/Client/List.hs`,`Cabal/src/Distribution/PackageDescription/Check/Target.hs`.